pkg/gator/reader/read_constraints.go

```diff
@@ -1,6 +1,7 @@
 package reader
```

Comment:
Seconding this! I was going to suggest that we throw the main logic in its own package. As I was looking for the right place, I started wondering if we wanted to pull the parsing logic (eventually) into the CF? Wdyt? @julianKatz

Reply:
> (eventually) into the CF?

From offline chatter, since the annotation lives around and defines K8s-specific concepts. So that's the reasons against adding the parsing in the CF framework.